crates/iota-graphql-rpc/src/types/query.rs

```diff
@@ -97,6 +98,21 @@ impl Query {
         type_args: Option<Vec<String>>,
         arguments: Option<Vec<serde_json::Value>>,
     ) -> Result<MoveViewResult> {
+        let chain_id_cache: &ChainIdentifierCache = ctx.data_unchecked();
+
+        let chain = chain_id_cache
+            .read(ctx.data_unchecked(), ctx.data_unchecked())
```

Comment:
something like this:
```suggestion
        let db = ctx.data_unchecked();
        let metrics = ctx.data_unchecked();
        let chain = chain_id_cache
            .read(db, metrics)
```